src/test/java/org/elasticsearch/common/xcontent/support/filtering/AbstractFilteringJsonGeneratorTests.java

```diff
@@ -0,0 +1,293 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.xcontent.support.filtering;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.ElasticsearchTestCase;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.Matchers.isEmptyString;
+import static org.hamcrest.Matchers.nullValue;
+
+public abstract class AbstractFilteringJsonGeneratorTests extends ElasticsearchTestCase {
+
+    protected abstract XContentType getXContentType();
+
+    protected abstract void assertXContentBuilder(String expected, XContentBuilder builder);
+
+    protected void assertString(String expected, XContentBuilder builder) {
+        assertNotNull(builder);
+        assertNotNull(expected);
+        assertNotNull(expected, not(isEmptyString()));
+
+        // Verify that the result is equal to the expected string
+        assertThat(builder.bytes().toUtf8(), Matchers.equalTo(expected));
+    }
+
+    protected void assertBinary(String expected, XContentBuilder builder) {
+        assertNotNull(builder);
+        assertNotNull(expected);
+        assertNotNull(expected, not(isEmptyString()));
+
+        try {
+            XContentParser jsonParser = XContentFactory.xContent(XContentType.JSON).createParser(expected.getBytes("UTF8"));
+            XContentParser xsonParser = XContentFactory.xContent(builder.contentType()).createParser(builder.bytes());
+
+            while (true) {
+                XContentParser.Token token1 = jsonParser.nextToken();
+                XContentParser.Token token2 = xsonParser.nextToken();
+                if (token1 == null) {
+                    assertThat(token2, nullValue());
+                    return;
+                }
+                assertThat(token1, Matchers.equalTo(token2));
+                switch (token1) {
+                    case FIELD_NAME:
+                        assertThat(jsonParser.currentName(), Matchers.equalTo(xsonParser.currentName()));
```

Comment:
static import of Matchers or just static import `Matchers.is()` (but thats my personal preference over `equalTo`...)